NMS-12372: Add gRPC support for IPC RPC/Sink (#2863)
OpenNMS/opennms: opennms-doc/guide-install/src/asciidoc/text/minion-grpc/steps-minion.adoc

==== Configure Minion

// No section numbers for step-by-step guide
:!sectnums:

> ===== Step 1: Disable ActiveMQ for RPC and Sink

Reviewer: ```suggestion
===== Step 1: Disable ActiveMQ for RPC and Sink.
```

.Disable ActiveMQ on _Minion_ startup
[source, shell]
----
cat <<EOF >${MINION_HOME}/etc/featuresBoot.d/disable-activemq.boot
!minion-jms
!opennms-core-ipc-rpc-jms
!opennms-core-ipc-sink-camel
EOF
----

> ===== Step 2: Enable GRPC for RPC and Sink

Reviewer: ```suggestion
===== Step 2: Enable GRPC for RPC and Sink.
```

[source, shell]
----
cat <<EOF >${MINION_HOME}/etc/featuresBoot.d/grpc.boot
opennms-core-ipc-grpc-client
EOF
----

> ===== Step 3: Configure gRPC server info

Reviewer: ```suggestion
===== Step 3: Configure gRPC server information. 
```

> .Add gRPC server for RPC/Sink communication

Reviewer: ```suggestion
.Add gRPC server for RPC/Sink communication.
```

[source, shell]
----
cat <<EOF >${MINION_HOME}/etc/org.opennms.core.ipc.grpc.client.cfg
host=localhost
port=8990
EOF
----

> ===== Step 4: Enable and configure TLS on gRPC client

Reviewer: ```suggestion
===== Step 4: Enable and configure TLS on gRPC client.
```

.Enable TLS and configure TLS certificates and private keys.
[source, shell]
----
cat <<EOF >${MINION_HOME}/etc/org.opennms.core.ipc.grpc.client.cfg
tls.enabled=true
trust.cert.filepath=/custom-path/ca.crt
client.cert.filepath=/custom-path/client.crt
client.private.key.filepath=/custom-path/client.pem
EOF
----

> ===== Step 4: Configure max message size if default 10MB is not sufficient.

Reviewer: ```suggestion
===== Step 5: Configure max. message size if default of 10MB is not sufficient.
```

> (needs to be configured both on server and client)

Reviewer: ```suggestion
(needs to be configured on both server and client)
```

> .Configure max message size

Reviewer: ```suggestion
.Configure max. message size
```

[source, shell]
----
cat <<EOF >${MINION_HOME}/etc/org.opennms.core.ipc.grpc.client.cfg
max.message.size=10485760
EOF
----

> ===== Step 5: Restart Minion to apply changes

Reviewer: ```suggestion
===== Step 6: Restart Minion to apply changes.
```

[source, shell]
----
systemctl restart minion
----

> ===== Step 6: Verify GRPC configuration and connectivity

Reviewer: ```suggestion
===== Step 7: Verify GRPC configuration and connectivity.
```

> .Login to Karaf Shell

Reviewer: ```suggestion
.Login to Karaf shell
```

[source, shell]
----
ssh admin@localhost -p 8201
----

.Test if gRPC client can connect to _{opennms-product-name}_ gRPC server
[source, shell]
----
feature:list | grep opennms-core-ipc-grpc-client
opennms-core-ipc-grpc-client                │ 26.0.0.SNAPSHOT  │ x        │ Started
----

.Test connectivity to Kafka
[source, shell]
----
opennms-health:check
Verifying the health of the container

Connecting to OpenNMS ReST API   [ Success  ]
Verifying installed bundles      [ Success  ]
Connecting to gRPC IPC Server    [ Success  ]

=> Everything is awesome
----

> ===== Step 7. Verify Minion functionality

Reviewer: ```suggestion
===== Step 8. Verify Minion functionality.
```
